SCPlayer/SCPlayer/Controllers/Cores/Home Screen/Home Screen Cell/Home List Track/HomeListTrackCollectionViewCell.swift

```diff
@@ -32,22 +30,58 @@ final class HomeListTrackCollectionViewCell: UICollectionViewCell, NibReusable {
             $0.register(cellType: HomeTrackCollectionViewCell.self)
         }
     }
+    
+    public func getData (selfGenre: String, listTrack: [Track]) {
+        self.listAllTrack = listTrack
+        self.selfGenre = selfGenre
+        loadData()
+    }
+    
+    private func loadData() {
+        if self.selfGenre == "All Songs" {
+            self.listTrackForGenre = self.listAllTrack
+        }
+        else {
+            self.listTrackForGenre.removeAll()
+            for item in self.listAllTrack {
+                let genre = item.genre ?? "NIL"
+                if genre == self.selfGenre && genre != "" && genre != "NIL" {
+                    self.listTrackForGenre.append(item)
+                }
+            }
+        }
+        DispatchQueue.main.async {
+            self.listTrackCollectionView.reloadData()
+        }
+    }
+    
 }
 
 extension HomeListTrackCollectionViewCell: UICollectionViewDataSource {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        return listTrack.count
+        return listTrackForGenre.count
     }
     
     func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = collectionView.dequeueReusableCell(for: indexPath, cellType: HomeTrackCollectionViewCell.self)
+
+        guard let nameOfSong = listTrackForGenre[indexPath.row].title else {
+            return cell
+        }
+        guard let artist = listTrackForGenre[indexPath.row].user?.username else {
+            return cell
+        }
+        guard let avatarUrlString = listTrackForGenre[indexPath.row].user?.avatarUrl else {
             return cell
+        }
+        cell.configureCell(imageURL: avatarUrlString, songNameLabel: nameOfSong, artistLabel: artist)
+        return cell
```

Comment:
```suggestion
        guard let nameOfSong = listTrackForGenre[indexPath.row].title,
                   let artist = listTrackForGenre[indexPath.row].user?.username,
                   let avatarUrlString = listTrackForGenre[indexPath.row].user?.avatarUrl else {
            return UITableViewCell()
        }
        let cell = collectionView.dequeueReusableCell(for: indexPath, cellType: HomeTrackCollectionViewCell.self)
        cell.configureCell(imageURL: avatarUrlString, songNameLabel: nameOfSong, artistLabel: artist)
        return cell
```
Với cả sao không truyền model kiểu Track vào luôn hàm configureCell em nhỉ ? Thay vì phải tạo hàm với 3 params tnay

Reply:
Vậng ạ, đoạn này em ngố quá @@ thấy nó cần 3 param đấy nên em truyển 3 param làm phức tạp thêm. truyền luôn track vào cho đơn giản @@